File: kotlin/matrix/src/main/kotlin/Matrix.kt
class Matrix(private val matrixAsString: String) {

    private val data: List<List<Int>> = matrixAsString.split("\n").map {
        it.trim().split("[ ]+".toRegex()).map {it.toInt()}
    }

    fun column(colNr: Int): List<Int> {
        return data.map {it[colNr-1]}.toList()
    }

    fun row(rowNr: Int): List<Int> {
        return data[rowNr-1].toList()
    }
}

File: typescript/matrix/matrix.ts
class Matrix {
  rows: number[][] = []
  columns: number[][] = []

  constructor(matrixString: string) {
    this.rows = matrixString
        .split("\n")
        .map(rowString =>
            rowString
                .split(/\s+/)
                .map(n => Number(n))
        )

    this.columns = this.rows[0]
        .map((_, colIndex) =>
          this.rows.map(row => row[colIndex])
        )
  }
}

export default Matrix
